src/components/sign-attributes/sign-attribute-modal.vue

```diff
@@ -0,0 +1,264 @@
+import SignAttributeBadge from './sign-attribute-badge.vue';
+// import ErrorService from '@/services/error';
+
+@Component({
+    name: 'sign-attribute-modal',
+    components: {
+        'sign-attribute-badge': SignAttributeBadge,
+    },
+})
+export default class SignAttributeModal extends Vue {
+    private comment: string = '';
+    private selected: string | null = null;
+
+    private mounted() {
+        this.setComment(this.attribute);
+    }
+
+    private get attribute() {
+        return this.$state.artefactEditor.selectedAttribute;
+    }
+
+    private get deleteAllowed() {
+        return this.attributeMetadata?.removable || false;
+    }
+
+    private get editAllowed() {
+        if (this.isMultiSelect) {
+            return this.attributeMetadata?.batchEditable || false;
+        } else {
+            return this.attributeMetadata?.editable || false;
+        }
+    }
+
+    private get attributeMetadata() {
+        if (!this.attribute) {
+            return undefined;
+        }
+        return this.$state.editions.current?.attributeMetadata?.getAttribute(
+            this.attribute.attributeId
+        );
+    }
+
+    private get possibleAttributeValues() {
+        if (!this.attribute || !this.attributeMetadata) {
+            console.warn(
+                "Can't return possible values if there is no attribute or metedata"
+            );
+            return [];
+        }
+        let values = this.attributeMetadata.values;
+
+        // Remove the values that are selected by other attributes of the same id
+        for (const si of this.$state.artefactEditor
+            .selectedSignsInterpretation) {
+            for (const attr of si.attributes.filter(
+                (a) => a.attributeId === this.attribute!.attributeId
+            )) {
+                values = values.filter((v) => v.id !== attr.attributeValueId);
+            }
+        }
+
+        return values;
+    }
+
+    private get isMultiSelect() {
+        return (
+            this.$state.artefactEditor.selectedSignsInterpretation.length !== 1
+        );
+    }
+
+    private get description(): string {
+        const metadata = this.$state.editions.current!.attributeMetadata!;
+        const attrMetadata = metadata.getAttribute(this.attribute!.attributeId);
+
+        if (!attrMetadata) {
+            return '';
+        }
+
+        let description = attrMetadata.description || '';
+
+        if (this.attribute!.attributeValueString !== 'TRUE') {
+            // A non-boolean attribute
+            const valueMetadata = metadata.getAttributeValue(
+                this.attribute!.attributeId,
+                this.attribute!.attributeValueId
+            );
+            if (valueMetadata && valueMetadata.description) {
+                description += valueMetadata.description;
+            }
+        }
+
+        return description;
+    }
+
+    @Watch('attribute')
+    private onAttributeChanged(
```

Comment:
So, I may not be following this correctly, but from what I see, if `this.$state.artefactEditor.selectedAttribute` changes, this code gets called, which will run `this.setComment` to set the comment for the new attribute.  It is not clear to me why we need a watcher here though.  Couldn't we just make a computed property `private get comment() {return this.attribute?.commentary?.commentary ?? ''}`?  I thought the reactivity would follow the chain of computed property getters, but maybe I am wrong, or maybe you are doing something different.

Reply:
I did not want SignalR updates of the comment to overwrite the text box, as this is *really* irritating.